CRM/Contribute/BAO/Contribution.php

```diff
@@ -3356,8 +3356,10 @@ public static function updateFinancialAccounts(&$params, $context = NULL, $skipT
     ) {
       return;
     }
-    if (($previousContributionStatus == 'Partially paid')
-      && $params['contribution']->contribution_status_id == array_search('Completed', $contributionStatus)
+    if ((($previousContributionStatus == 'Partially paid'
+      && $params['contribution']->contribution_status_id == array_search('Completed', $contributionStatus))
```

Comment:
This could be made easier to read - ideally it would look like

if ($currentContributionStatus == 'Completed' && in_array($previousContributionStatus, array('Partially paid', 'Pending') {

}